deepchecks/nlp/utils/text_data_plot.py

```diff
@@ -0,0 +1,201 @@
+
+    mean = data.mean()
+    percentile_90 = data.quantile(0.9)
+    percentile_10 = data.quantile(0.1)
+    median = data.median()
+
+    trace = go.Scatter(x=x_value, y=y_value, fill='tozeroy', showlegend=False,
+                       hovertemplate=f'<b>{property_name}:</b> ''%{x}<br><b>Density:</b> %{y}<extra></extra>',
+                       line={'color': feature_distribution_colors['feature'],
+                             'shape': 'linear', 'width': 5})
+    shapes = []
+    annotations = []
+
+    shapes.append(dict(type='line', x0=mean, y0=0, x1=mean, y1=max(y_value),
+                       line={'color': feature_distribution_colors['measure'], 'dash': 'dash', 'width': 3}))
+    mean_xpos = mean + max(x_value) * 0.02 if median < mean else mean - max(x_value) * 0.02
+    annotations.append(dict(x=mean_xpos, y=max(y_value)/2, text='<b>Mean</b>', showarrow=False,
+                            textangle=-90, font={'size': 12}))
+
+    shapes.append(dict(type='line', x0=median, y0=0, x1=median, y1=max(y_value),
+                       line={'color': feature_distribution_colors['measure'], 'dash': 'dot', 'width': 3}))
+    median_xpos = median - max(x_value) * 0.02 if median < mean else median + max(x_value) * 0.02
+    annotations.append(dict(x=median_xpos, y=max(y_value)/2, text='<b>Median</b>', showarrow=False,
+                            textangle=-90, font={'size': 12}))
+
+    shapes.append(dict(type='line', x0=percentile_10, y0=0, x1=percentile_10, y1=max(y_value),
+                       line={'color': feature_distribution_colors['measure'], 'dash': 'dashdot', 'width': 3}))
+    annotations.append(dict(x=percentile_10 - max(x_value)*0.02, y=max(y_value)/2, textangle=-90,
+                            text='<b>10<sup>th</sup> Percentile</b>', showarrow=False, font={'size': 12}))
+
+    shapes.append(dict(type='line', x0=percentile_90, y0=0, x1=percentile_90, y1=max(y_value),
+                       line={'color': feature_distribution_colors['measure'], 'dash': 'dashdot', 'width': 3}))
+    annotations.append(dict(x=percentile_90 + max(x_value)*0.02, y=max(y_value)/2, textangle=-90,
+                            text='<b>90<sup>th</sup> Percentile</b>', showarrow=False, font={'size': 12}))
+
+    xaxis_layout = dict(title=property_name)
+    yaxis_layout = dict(title='Density')
+
+    return trace, shapes, annotations, xaxis_layout, yaxis_layout
+
+
+def text_data_describe_plot(all_properties_data, n_samples, label, categorical_metadata, numerical_metadata,
+                            categorical_properties, numerical_properties, properties: List[str]):
+    """Return a plotly figure instance.
+
+    Parameters
+    ----------
+    all_properties_data:
+    n_samples:
+    label:
+    categorical_metadata:
+    numerical_metadata:
+    categorical_properties:
+    numerical_properties:
+    properties : List[str]
+        List of property names to consider for generating property distribution graphs.
```

Comment:
Missing parameter descriptions

Reply:
Yeah I have to add the docstring for all the functions.